druid/src/core.rs

```diff
@@ -272,23 +279,21 @@ impl<T, W: Widget<T>> WidgetPod<T, W> {
     /// [`Rect`]: struct.Rect.html
     /// [`Size`]: struct.Size.html
     /// [`LifeCycle::Size`]: enum.LifeCycle.html#variant.Size
-    pub fn set_origin(&mut self, ctx: &mut LayoutCtx, data: &T, env: &Env, origin: Point) {
-        self.state.origin = origin;
+    pub fn set_origin<'a>(
+        &mut self,
+        ctx: &mut impl ChangeCtx<'a>,
+        _data: &T,
+        _env: &Env,
+        origin: Point,
+    ) {
+        //TODO: decide whether we should keep data and env for compatibility or do a breaking change
         self.state.is_expecting_set_origin_call = false;
-        let layout_rect = self.layout_rect();
-
-        // if the widget has moved, it may have moved under the mouse, in which
-        // case we need to handle that.
-        if WidgetPod::set_hot_state(
-            &mut self.inner,
-            &mut self.state,
-            ctx.state,
-            layout_rect,
-            ctx.mouse_pos,
-            data,
-            env,
-        ) {
-            ctx.widget_state.merge_up(&mut self.state);
+
+        if origin != self.state.origin {
+            self.state.origin = origin;
+            self.state.view_context_changed = true;
+            // identical to calling merge up but faster!
+            ctx.state().widget_state.children_view_context_changed = true;
```

Comment:
Definitely not *identical* as for example it doesn't do invalidation region handling. Anyway `merge_up()` will be called higher up the stack anyway right? Generally speaking you would call `set_origin()` from somewhere like `layout()` and when that stack unwinds `WidgetPod` will call `merge_up()`. Is there a reasonable scenario where you can call `set_origin()` outside of such a context?

If not then I'm thinking we can just delete this line of code and the `merge_up()` higher up the stack will deal with propagation just fine.

Reply:
It is identical in its effect on the parent's state, since `view_context_changed` is the only part of the state which changed and merged up.

> Anyway merge_up() will be called higher up the stack anyway right?

Let's again say we have the following hierarchy of widgets: A is parent of B and B is parent of C. 
Now A calls `B.layout(...)`, in this call B calls `C.layout(...)` in the end the states of B and C are merged.
Then B calls `C.set_origin(...)`. 
Then B.layout(...) ends and B and A are merged, which does not take C's state into account.
Therefore if we don't pass `ctx` in `C.set_origin`, there is no way to merge up the state, since there is no guarantee, that any method of C is called again before paint.

Reply:
Thanks for the explanation. I was wrong and you are correct, we need this. (*I was fooled by examples still seeming to work, but that was a mirage.*)

I still don't like the *identical* claim and let me explain a bit more of why. So it's not even so much about what is happening today - it is about how brittle this comment is. When someone will update `merge_up()` in the future to do more they probably won't update this comment. Which means that there will be an out-of-sync comment, if you will.

The better approach is to have the comments describe things that are actually happening in this section, and not be dependant on some more distant piece of code not changing.

On the topic of brittleness and this being a performance trick, in the past we've actually deliberately used `merge_up()` while knowing it has a performance cost. The reason being that the win is a more robust function that won't break as easily as new functionality is added.

Now where exactly to draw the line is subjective of course. I personally have a tolerance for more performance tweaks than just always calling `merge_up()` which is also why I think it's fine if we keep this trick in. However I would prefer to change the wording of the comment.

Perhaps something like:
```rust
// Instead of calling merge_up we directly propagate the flag we care about, to gain performance
```

Reply:
merge_up() is great for methods which call into the widget trait. The difference here is imo that it is obvious which parts of the state are getting changed by calling `set_origin()`, but i'm completely fine with using `merge_up` instead for consistency. 

I will add the comment for now.